### ISSUE TITLE
Error: no JSON at incomin...

### ISSUE THREAD

**Issue by jensrossbach** [586837r/node-red-contrib-alexa-remote2#84]
I'm using alexa-remote2 Echo node to activate Bluetooth connection to my soundbar upon a certain trigger (in my case when Echo starts playing music). This worked fine in the beginning but suddenly stopped working and I get an error message "Error: no JSON at incomin..." (cannot see whole message as it is cut at the end).

I'm using proxy authentication with seems to be fine. The error is available in the latest version (3.7.0).

**Comment by jensrossbach** [comment]
Additional information: When I restart my flows, everything works again for some time, but then some time later, the error reappears.

**Comment by bwims** [comment]
I have found that this error occurs for me when I send multiple instances of the same command, so that, e.g. I try to disconnect bluetooth when it is already disconnected.  Is it possible you are doing this under certain conditions?

**Comment by jensrossbach** [comment]
@bwims yes that could be the case (however I have no proof for this). Still I would consider this a bug.

**Comment by bwims** [comment]
I use the RBE node to block repeated commands unless the payload changes.

**Comment by jensrossbach** [comment]
I've meanwhile changed my flow to avoid sending the same command subsequently to the alexa nodes but the error still occurs. Seems to be less often now but still happening.

**Comment by Myzhar** [comment]
Have you found a solution to this problem? I continue to have random "NO JSON" error 

This is the flow that I'm testing:
![Screenshot from 2020-04-30 14-54-56](https://user-images.githubusercontent.com/3648617/80712676-9a14e200-8af2-11ea-9574-e691a1582907.png)

Often the last node gets the `NO JSON` error, but sometimes it happens also for the others

**Comment by Myzhar** [comment]
Please update the issue if you find something new

**Comment by bwims** [comment]
will do!

**Comment by tekybrit** [comment]
I have not seen this error since I inserted a delay node to limit the message rate.
This leaves Alexa enough time to complete one message before she gets the next message.
In my use case, limiting the message rate to one every 3 seconds works fine.

**Comment by jensrossbach** [comment]
I don’t think that there are so high frequent requests to the echo nodes in my flow. I’m just switching on Bluetooth connection when the echo starts playing music and switching off Bluetooth connection one minute after the echo stops playing music. Still I get this error from time to time.

**Comment by Myzhar** [comment]
I tried suddenly with delays, but they are not useful. What instead helped me is using a `catch` node that catches the error and triggers the node again after a little delay.
I add a catch node for each Amazon node.

**Comment by jensrossbach** [comment]
If this actually helps, then it should be implemented in the nodes themselves as a recovery mechanism.

**Comment by 586837r** [comment]
Amazon sends a `"Rate exceeded: Too many requests."` in these cases. I'm thinking about a way to handle this generally. (And to have more helpful errors.) 
Also keep in mind that if possible you can also send a command sequence at once with the `Node` option instead of chaining them.

**Comment by Myzhar** [comment]
@586837r this is interesting. I will investigate how to do it

**Comment by adampetrovic** [comment]
> Also keep in mind that if possible you can also send a command sequence at once with the Node option instead of chaining them.
@586837r do you have an example on how we might do that?

**Comment by nicedevil007** [comment]
What is the actual status of this issue?  I still get this error.
@Myzhar How did you solve this with the catch node?

In my case I just want to set the volume to a special amount.

If the error was triggered I can't get anything beside the error itself. So how did you reenter the given value until it is sent successfully?

**Comment by obaldius** [comment]
What has solved the problem to me is using de Node tool whithin the Alexa Routine Node instead of branching the flow in Node Red itself. You can choose between Serial or Parallel execution whithin that action

**Comment by obaldius** [comment]
I must add, I believe it's just a "Too Many Requests" error. If you chain several Alexa nodes in serial execution, putting a delay of 1.5 - 2 seconds between them should solve the issue. Putting them in parallel greatly increases the risk of such error. For parallel request, better use the Node setting whithin the Routine Node
